libtock/ble.c

```diff
@@ -10,41 +10,65 @@
 #include <string.h>
 
 int ble_start_advertising(int pdu_type, uint8_t* advd, int len, uint16_t interval) {
-  int err = allow(BLE_DRIVER_NUMBER, 0, advd, len);
-  if (err < TOCK_SUCCESS)
-    return err;
+  allow_ro_return_t err = allow_readonly(BLE_DRIVER_NUMBER, BLE_CFG_ADV_BUF_ALLOWRO, advd, len);
+  if (!err.success)
+    return tock_error_to_rcode(err.error);
 
-  return command(BLE_DRIVER_NUMBER, BLE_ADV_START_CMD, pdu_type, interval);
+  syscall_return_t res = command2(BLE_DRIVER_NUMBER, BLE_ADV_START_CMD, pdu_type, interval);
+  if (res.type == TOCK_SYSCALL_SUCCESS_U32) {
+    return res.data[0];
```

Comment:
```suggestion
  if (res.type == TOCK_SYSCALL_SUCCESS) {
    return TOCK_SUCCESS;
```

`TOCK_SYSCALL_SUCCESS_U32` will only be present when the capsule returns `CommandValue::success_u32()`, and looking at the updated capsule it does not seem to do that anywhere

Reply:
This applies to all the command calls

Reply:
Of course! Done